proposals/53:kread-start/core-eval-support.js

```diff
@@ -0,0 +1,152 @@
+// @ts-check
+import {
+  Far,
+  makeMarshal,
+  makeTranslationTable,
+} from '../../upgrade-test-scripts/lib/unmarshal.js';
+import { Fail, NonNullish } from '../../upgrade-test-scripts/lib/assert.js';
+
+// TODO: factor out ambient authority from these
+// or at least allow caller to supply authority.
+import { mintIST } from '../../upgrade-test-scripts/lib/econHelpers.js';
+import { agoric } from '../../upgrade-test-scripts/lib/cliHelper.js';
+
+// move to unmarshal.js?
+const makeBoardUnmarshal = () => {
+  const synthesizeRemotable = (_slot, iface) =>
+    Far(iface.replace(/^Alleged: /, ''), {});
+
+  const { convertValToSlot, convertSlotToVal } = makeTranslationTable(
+    slot => Fail`unknown id: ${slot}`,
+    synthesizeRemotable,
+  );
+
+  return makeMarshal(convertValToSlot, convertSlotToVal);
+};
+
+export const getContractInfo = async (path, io = {}) => {
+  const m = makeBoardUnmarshal();
+  const {
+    agoric: { follow = agoric.follow },
+    prefix = 'published.',
+  } = io;
+  console.log('@@TODO: prevent agoric follow hang', prefix, path);
+  const txt = await follow('-lF', `:${prefix}${path}`, '-o', 'text');
+  const { body, slots } = JSON.parse(txt);
+  return m.fromCapData({ body, slots });
+};
+
+// not really core-eval related
+export const testIncludes = (t, needle, haystack, label, sense = true) => {
+  t.log(needle, sense ? 'in' : 'not in', haystack.length, label, '?');
+  const check = sense ? t.deepEqual : t.notDeepEqual;
+  if (sense) {
+    t.deepEqual(
+      haystack.filter(c => c === needle),
+      [needle],
+    );
+  } else {
+    t.deepEqual(
+      haystack.filter(c => c === needle),
+      [],
+    );
+  }
+};
```

Comment:
I find this abstraction confusing. Why not just something like,
```js
haystacks.includes(needle);
```

Reply:
yeah; that's kind of a mess. I think the motivation was: to get more than a 1 bit report when things go wrong.

Reply:
```suggestion
/**
 * Asserts that `haystack` includes `needle` (or when `sense` is false, that it
 * does not), providing pretty output in the case of failure.
 *
 * @param {import('ava').ExecutionContext} t
 * @param {unknown} needle
 * @param {unknown[]} haystack
 * @param {string} label
 * @param {boolean} [sense] true to assert inclusion; false for exclusion
 * @returns {void}
 */
export const testIncludes = (t, needle, haystack, label, sense = true) => {
  const matches = haystack.filter(c => Object.is(c, needle));
  t.deepEqual(matches, sense ? [needle] : [], label);
};
```

I'd also recommend reordering parameters so invocations read like `testIncludes(t, haystack, needle, label)`.